### PR TITLE
Added pod annotation with job RetryCount

### PR DESCRIPTION
**Pull request by belo4ya**
Added pod annotation with current Job RetryCount version: `volcano.sh/job-retry-count=<Job.Status.RetryCount>`

feat #3528

### CODE REVIEW

File: go.mod
Comment (belo4ya) [reply go.mod:175]: I'm not sure, I don't quite understand the process of making changes simultaneously in two repositories.
But to implement this feature, I need these changes: https://github.com/volcano-sh/apis/pull/128. Once they are merged into master/tag, we can remove this replace.
Therefore, this PR is blocked by PR: https://github.com/volcano-sh/apis/pull/128.

Comment (Monokaix) [reply go.mod:175]: Ok, I see, will merge it first: )

Comment (Monokaix) [reply go.mod:175]: Hi, https://github.com/volcano-sh/apis/pull/128 has been merged, you can change go mod and rebase your code.

File: go.mod
Comment (belo4ya) [inline go.mod:174]: I couldn't do it without adding this line(

```bash
go get github.com/volcano-sh/apis@master
go: github.com/volcano-sh/apis@master (v1.10.0-alpha.0.0.20240709084748-78d912ce096c) requires github.com/volcano-sh/apis@v1.10.0-alpha.0.0.20240709084748-78d912ce096c: parsing go.mod:
        module declares its path as: volcano.sh/apis
                but was required as: github.com/volcano-sh/apis
```

Perhaps this is due to the fact that the changes have been included in the master, but have not yet been included in any tag.

As far as I understand, I still have to wait for the next tag to appear in [volcano-sh/apis](https://github.com/volcano-sh/apis) and then make changes again?

File: go.mod
Comment (belo4ya) [reply go.mod]: I couldn't do it without adding this line - https://github.com/volcano-sh/volcano/pull/3544#discussion_r1670347637

File: go.mod
Comment (belo4ya) [reply go.mod]: Thanks, it worked

**Review by hwdef — COMMENTED** [review]
/approve

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**
